### DISCUSSION TITLE
Autocompletion for the arguments of the selected function provided by LSP

### DISCUSSION THREAD

**Discussion by Chen-Yulin** [hrsh7th/nvim-cmp#1317] [Q&A] [answered]
In the demo video：
https://user-images.githubusercontent.com/629908/139000570-3ac39587-a88b-43c6-b35e-207489719359.mp4
I seems the argument of the selected function will appear after the function completion is confirmed, and enter SELECT mode, just activing like a snip.
I was wondering if this is nvim-cmp's feature or other plugin's, since after confirming the function completion, only the name of the function is completed.

**Comment by Chen-Yulin** [comment, marked as answer]
I use clangd as my language server.

**Reply by hrsh7th** [reply]
See clangd's command arguments.

https://github.com/llvm/llvm-project/blob/1f914944b6c9a5e4229ceb9f06140fdf178c5ea0/clang-tools-extra/clangd/tool/ClangdMain.cpp#L243

You can add `--function-arg-placeholders`.

**Reply by Chen-Yulin** [reply]
Yeah, I think that's the point, but can you be more specific, such as how to setup the clangd server with the arg `--function-arg-placeholders`.
I am sorry that I am fairly new to nvim and lsp configuration.

**Reply by hrsh7th** [reply]
I don't know but I'd investigate.
Next time, please investigate by yourself and teach some beginner.

https://github.com/caoTayTang/NvChad_dotfiles/blob/a49d2ec361198351058f11612621273f3eb1614d/lua/custom/plugins/lspconfig.lua#L15

**Reply by Chen-Yulin** [reply]
Thanks a lot, I will pass it on!
